mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1245,6 +1249,1321 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
+
+    auto dilationsAttr = op->template getAttrOfType<ArrayAttr>("dilations");
+    auto stridesAttr = op->template getAttrOfType<ArrayAttr>("strides");
+    auto paddingAttr = op->template getAttrOfType<ArrayAttr>("padding");
+
+    // Get shape of filter tensor.
+    auto filterType = op.filter().getType().template dyn_cast<MemRefType>();
+    auto filterShape = filterType.getShape();
+    auto filterElementType = filterType.getElementType();
+
+    // Get shape of input tensor.
+    auto inputType = op.input().getType().template dyn_cast<MemRefType>();
+    auto inputShape = inputType.getShape();
+    auto inputElementType = inputType.getElementType();
+
+    // Get shape of output tensor.
+    auto outputType = op.output().getType().template dyn_cast<MemRefType>();
+    auto outputShape = outputType.getShape();
+    auto outputElementType = outputType.getElementType();
+
+    // Obtain convolution parameters: padding / dialtion / stride.
+    auto leftPadH =
+        paddingAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto leftPadW =
+        paddingAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationH =
+        dilationsAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationW =
+        dilationsAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto strideH =
+        stridesAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto strideW =
+        stridesAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+
+    // get y, x, ho, wo, hi, wi
+    int64_t g, n, k, c, y, x, ho, wo, hi, wi;
+    g = n = k = c = y = x = ho = wo = hi = wi = 0;
+    for (unsigned i = 0; i < filterLayoutAttr.size(); ++i) {
+      auto filterAttr =
+          filterLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto inputAttr =
+          inputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto outputAttr =
+          outputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+
+      if (filterAttr.getValue() == "g") {
+        g = filterShape[i];
+      } else if (filterAttr.getValue() == "k") {
+        k = filterShape[i];
+      } else if (filterAttr.getValue() == "c") {
+        c = filterShape[i];
+      } else if (filterAttr.getValue() == "y") {
+        y = filterShape[i];
+      } else if (filterAttr.getValue() == "x") {
+        x = filterShape[i];
+      }
+
+      if (inputAttr.getValue() == "ni") {
+        n = inputShape[i];
+      } else if (inputAttr.getValue() == "hi") {
+        hi = inputShape[i];
+      } else if (inputAttr.getValue() == "wi") {
+        wi = inputShape[i];
+      }
+
+      if (outputAttr.getValue() == "ho") {
+        ho = outputShape[i];
+      } else if (outputAttr.getValue() == "wo") {
+        wo = outputShape[i];
+      }
+    }
+
+    // compute padding hi/wi.
+    auto hiPadded = 1 + (y - 1) * dilationH + (ho - 1) * strideH;
```

Comment:
This padding formula had problems
Please refer to latest code

Reply:
done